### ISSUE TITLE
[TT-1313] Webhook event for Circuit Breakers doesn't work

### ISSUE THREAD

**Issue by ilijabojanovic** [TykTechnologies/tyk#1935]
**Do you want to request a *feature* or report a *bug*?**
BUG
**What is the current behavior?**
When we enable webhook for Circuit Breakers plugin and simulate Circuit Breaker, webhook event doesn't fire
**What is the expected behavior?**
Webhook should fire for Breaker Reset and Breaker Tripped events.
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**
1. Create webhook
2. Load api definition
3. Generate some traffic to /status/500 endpoint of this api

Api definition:
```json
{
    "id": "5bbfaba49b7d7c093aebc769",
    "name": "webhook",
    "slug": "webhook",
    "api_id": "efe0d21ed9a64ffd6d27a4ee48e13557",
    "org_id": "5bbc7e4e9b7d7c069575e2ac",
    "use_keyless": true,
    "use_oauth2": false,
    "use_openid": false,
    "openid_options": {
        "providers": [],
        "segregate_by_client": false
    },
    "oauth_meta": {
        "allowed_access_types": [],
        "allowed_authorize_types": [],
        "auth_login_redirect": ""
    },
    "auth": {
        "use_param": false,
        "param_name": "",
        "use_cookie": false,
        "cookie_name": "",
        "auth_header_name": "",
        "use_certificate": false
    },
    "use_basic_auth": false,
    "basic_auth": {
        "disable_caching": false,
        "cache_ttl": 0
    },
    "use_mutual_tls_auth": false,
    "client_certificates": [],
    "upstream_certificates": {},
    "pinned_public_keys": {},
    "enable_jwt": false,
    "use_standard_auth": false,
    "enable_coprocess_auth": false,
    "jwt_signing_method": "",
    "jwt_source": "",
    "jwt_identity_base_field": "",
    "jwt_client_base_field": "",
    "jwt_policy_field_name": "",
    "jwt_disable_issued_at_validation": false,
    "jwt_disable_expires_at_validation": false,
    "jwt_disable_not_before_validation": false,
    "jwt_skip_kid": false,
    "notifications": {
        "shared_secret": "",
        "oauth_on_keychange_url": ""
    },
    "enable_signature_checking": false,
    "hmac_allowed_clock_skew": -1,
    "base_identity_provided_by": "",
    "definition": {
        "location": "header",
        "key": "x-api-version",
        "strip_path": false
    },
    "version_data": {
        "not_versioned": true,
        "default_version": "",
        "versions": {
            "Default": {
                "name": "Default",
                "expires": "",
                "paths": {
                    "ignored": [],
                    "white_list": [],
                    "black_list": []
                },
                "use_extended_paths": true,
                "extended_paths": {
                    "circuit_breakers": [
                        {
                            "path": "/status/5000",
                            "method": "GET",
                            "threshold_percent": 10,
                            "samples": 2,
                            "return_to_service_after": 60
                        }
                    ]
                },
                "global_headers": {},
                "global_headers_remove": [],
                "global_size_limit": 0,
                "override_target": ""
            }
        }
    },
    "uptime_tests": {
        "check_list": [],
        "config": {
            "expire_utime_after": 0,
            "service_discovery": {
                "use_discovery_service": false,
                "query_endpoint": "",
                "use_nested_query": false,
                "parent_data_path": "",
                "data_path": "",
                "port_data_path": "",
                "target_path": "",
                "use_target_list": false,
                "cache_timeout": 60,
                "endpoint_returns_list": false
            },
            "recheck_wait": 0
        }
    },
    "proxy": {
        "preserve_host_header": false,
        "listen_path": "/webhook/",
        "target_url": "http://httpbin.org/",
        "strip_listen_path": true,
        "enable_load_balancing": false,
        "target_list": [],
        "check_host_against_uptime_tests": false,
        "service_discovery": {
            "use_discovery_service": false,
            "query_endpoint": "",
            "use_nested_query": false,
            "parent_data_path": "",
            "data_path": "",
            "port_data_path": "",
            "target_path": "",
            "use_target_list": false,
            "cache_timeout": 0,
            "endpoint_returns_list": false
        },
        "transport": {
            "ssl_ciphers": [],
            "ssl_min_version": 0,
            "proxy_url": ""
        }
    },
    "disable_rate_limit": false,
    "disable_quota": false,
    "custom_middleware": {
        "pre": [],
        "post": [],
        "post_key_auth": [],
        "auth_check": {
            "name": "",
            "path": "",
            "require_session": false
        },
        "response": [],
        "driver": "",
        "id_extractor": {
            "extract_from": "",
            "extract_with": "",
            "extractor_config": {}
        }
    },
    "custom_middleware_bundle": "",
    "cache_options": {
        "cache_timeout": 60,
        "enable_cache": true,
        "cache_all_safe_requests": false,
        "cache_response_codes": [],
        "enable_upstream_cache_control": false,
        "cache_control_ttl_header": ""
    },
    "session_lifetime": 0,
    "active": true,
    "auth_provider": {
        "name": "",
        "storage_engine": "",
        "meta": {}
    },
    "session_provider": {
        "name": "",
        "storage_engine": "",
        "meta": {}
    },
    "event_handlers": {
        "events": {
            "BreakerReset": [
                {
                    "handler_name": "eh_web_hook_handler",
                    "handler_meta": {
                        "api_model": {},
                        "event_timeout": 60,
                        "header_map": {
                            "qa": "qa"
                        },
                        "id": "5bbc80f39b7d7c08eb9ed9ab",
                        "method": "POST",
                        "name": "Test",
                        "org_id": "5bbc7e4e9b7d7c069575e2ac",
                        "target_path": "https://webhook.site/7de9dd5f-dd33-4a08-ba04-760ea7d94fbf",
                        "template_path": ""
                    }
                }
            ],
            "BreakerTripped": [
                {
                    "handler_name": "eh_web_hook_handler",
                    "handler_meta": {
                        "api_model": {},
                        "event_timeout": 60,
                        "header_map": {
                            "qa": "qa"
                        },
                        "id": "5bbc80f39b7d7c08eb9ed9ab",
                        "method": "POST",
                        "name": "Test",
                        "org_id": "5bbc7e4e9b7d7c069575e2ac",
                        "target_path": "https://webhook.site/7de9dd5f-dd33-4a08-ba04-760ea7d94fbf",
                        "template_path": ""
                    }
                }
            ]
        }
    },
    "enable_batch_request_support": false,
    "enable_ip_whitelisting": false,
    "allowed_ips": [],
    "enable_ip_blacklisting": false,
    "blacklisted_ips": [],
    "dont_set_quota_on_create": false,
    "expire_analytics_after": 0,
    "response_processors": [],
    "CORS": {
        "enable": false,
        "allowed_origins": [],
        "allowed_methods": [],
        "allowed_headers": [],
        "exposed_headers": [],
        "allow_credentials": false,
        "max_age": 24,
        "options_passthrough": false,
        "debug": false
    },
    "domain": "",
    "do_not_track": false,
    "tags": [],
    "enable_context_vars": false,
    "config_data": {},
    "tag_headers": [],
    "global_rate_limit": {
        "rate": 0,
        "per": 0
    },
    "strip_auth_data": false
}
```
**Which versions of Tyk affected by this issue? Did this work in previous versions of Tyk?**
Master/Stable

**Comment by letzya** [comment]
Is this still an issue?

**Comment by sedkis** [comment]
This is still an issue

**Comment by andyo-tyk** [comment]
This was fixed in Tyk 4.2.0 / 4.0.5
